From 9e2674712261458c7828cf873d4582361e6c8165 Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Mon, 9 Sep 2024 13:27:25 +0100
Subject: [PATCH] Add cstdint to compression.cc (#755)

Something has updated in the default conda-forge installation (e.g. new compilers)
that means we are now hitting this.

Fixes #678.
---
 newsfragments/755.misc                | 1 +
 src/dxtbx/boost_python/compression.cc | 1 +
 2 files changed, 2 insertions(+)
 create mode 100644 newsfragments/755.misc

diff --git a/newsfragments/755.misc b/newsfragments/755.misc
new file mode 100644
index 00000000..3eefecb4
--- /dev/null
+++ b/newsfragments/755.misc
@@ -0,0 +1 @@
+Fix non-inclusion of cstdint in dxtbx sources.
diff --git a/src/dxtbx/boost_python/compression.cc b/src/dxtbx/boost_python/compression.cc
index 6957d1bd..0e92e492 100644
--- a/src/dxtbx/boost_python/compression.cc
+++ b/src/dxtbx/boost_python/compression.cc
@@ -1,6 +1,7 @@
 #include "compression.h"
 #include "dxtbx/error.h"
 #include <assert.h>
+#include <cstdint>
 
 typedef union {
   char b[2];
-- 
2.46.0

